addons/web/static/src/search/search_bar/search_bar.js

```diff
@@ -284,7 +284,7 @@ export class SearchBar extends Component {
         let domain = [];
         if (searchItem.domain) {
             try {
-                domain = new Domain(searchItem.domain).toList();
+                domain = new Domain(searchItem.domain).toList(this.env.searchModel.context);
```

Comment:
I would prefer to declare a
```js
const context = { ...this.env.searchModel.globalContext, ...field.context };
```
(after const field) and use it here and in the name_search. I don't think it is a good idea to use this.env.searchModel.context since it is variable